### ISSUE TITLE
Error Standardization: Configuration API

### ISSUE THREAD

**Issue by cicoyle** [dapr/dapr#7485]
Richer errors in Dapr for the **Configuration API**.

**Note**: Please clone this issue and assign the clone of this issue to yourself if you would like to contribute to the error standardization efforts.

# Scope of work

- [ ] Choose 5-10 errors in the Configuration API to enrich. Ideally, cover as many errors as possible from the API.
- [ ] Follow the steps outlined in [this README](https://github.com/dapr/dapr/blob/master/pkg/api/errors/README.md) for each error being updated

# Definition of done

- [ ] Update 5-10 errors in the Configuration API
- [ ] Update the unit tests, if necessary
- [ ] Write integration tests to cover each updated errors

# Keep in mind

Unless a major version is being released and there is a solid reason for introducing backwards incompatibility, we can't change the API, meaning the HTTP and gRPC status codes, and the error codes must stay the same. An exception to this is the gRPC `StatusCode.UNKNOWN` which should be converted to a specific status.

See the [kit errors pkg README](https://github.com/dapr/kit/blob/main/errors/README.md) for another usage example. For each error being updated, you can use the following helper functions to add error details, see the kit [errors pkg](https://github.com/dapr/kit/tree/main/errors) for more details:

- `WithHelpLink()`
-  `WithHelp()`
-  `WithErrorInfo()` - **Required**
-  `WithFieldViolation()`
-  `WithDetails()`

# Reference PRs
Existing error standardization for: 
- [State API](https://github.com/dapr/dapr/pull/7257)
- [PubSub API](https://github.com/dapr/dapr/pull/7322)

**Comment by stackanthony** [comment]
Hi @cicoyle, does this issue still need to be worked on? I haven't contributed to the project yet, but looking forward to working on this one, if possible.

**Comment by cicoyle** [comment]
> Hi @cicoyle, does this issue still need to be worked on? I haven't contributed to the project yet, but looking forward to working on this one, if possible.

Yes this is available to be worked on. I'm happy to hear you are interested in contributing to Dapr. Please clone this issue and assign yourself to it.

Please lmk if you need any assistance along the way 🙂

**Comment by stackanthony** [comment]
Forsure, will do. I'll need some time to get acclimated to the codebase. Will let you know if I have any questions.

FYI - I worked at HashiCorp too, awesome to see someone from there contributing to the project @cicoyle

**Comment by stackanthony** [comment]
When you say clone this issue, how exactly should I do this? I've never done something like cloning an issue, only a repo lol. Can't find anything online about it either, maybe I misunderstood.

**Comment by cicoyle** [comment]
@stackanthony - clone as in, create a new issue, copy/paste the content, then assign your name to it :) Ah cool - I was at HashiCorp for a while, and am very happy to be at Diagrid now

**Comment by stackanthony** [comment]
Cool, just did it. Referenced in #7785

**Comment by ogulcanaydogan** [comment]
I'd like to work on this. Based on the State API (#7257) and PubSub API (#7322) reference PRs, I'll create `pkg/api/errors/configuration.go` following the same pattern and update the HTTP/gRPC handlers. I'll clone this issue and assign it to myself.

**Comment by ogulcanaydogan** [comment]
I'd like to take this one. I'll follow the `state.go` pattern and migrate the Configuration API errors (store not configured, store not found, get, subscribe, unsubscribe) to dedicated rich-error builders in `pkg/api/errors/configuration.go`, adding the missing gRPC reason codes and updating the gRPC and HTTP call sites.
